__test__/pet.test.js

```diff
@@ -0,0 +1,132 @@
+const Pet = require ('../src/pet.js')
+
+describe('constructor', () => {
+    it('returns an object', () => {
+        expect(new Pet ('Bob')).toBeInstanceOf(Object);
+    });
+    it('sets the name property', () => {
+        const pet = new Pet('Bob');
+        expect(pet.name).toEqual('Bob');
+      });
+    it('has an initial age of 0', () => {
+      const pet = new Pet('Bob')
+      pet.growUp()
+      expect(pet.age).toEqual(1);
+    })
+    it('has inital hunger of 0', () => {
+      const pet = new Pet ('Bob')
+      pet.getHungry()
+      expect(pet.hunger).toEqual(5)
+    })
+    it('has inital fitness of 10', () => {
+      const pet = new Pet ('Bob')
+      pet.becomeUnfit()
+      expect(pet.fitness).toEqual(7)
+    })
+    });
+
+    describe('walk', () => {
+      it('increases fitness by 4', () => {
+        const pet = new Pet('Bob')
+        pet.fitness = 4;
+        pet.walk()
+        expect(pet.fitness).toEqual(8)
+      })
+      it('it increases fitness to a maximum of 10', () => {
+        const pet = new Pet ('Bob')
+        pet.fitness = 8;
+        pet.walk()
+        expect(pet.fitness).toEqual(10)
+      })
+      it('it throws an error if the pet is not alive', () => {
+        const pet = new Pet ('Bob')
+        pet.age = 30
+        expect(() => pet.walk()).toThrow('Your pet is no longer alive')
+      })
+    })
+
+    describe('grow up', () => {
+      it('increases the age of the pet in incrememnts of 1', () => {
+        const pet = new Pet('Bob');
+        pet.growUp();
+        expect(pet.age).toEqual(1);
+      });
+      it('it throws an error if pet is no longer alive', () => {
+        const pet = new Pet('Bob');
+        pet.age = 30;
+        expect(() => pet.growUp()).toThrow('Your pet is no longer alive');
+      });
+    });
+
+    describe('feed', () => {
+      it('it prevents hunger level from going below 0', () => {
+        const pet = new Pet ('Bob')
+        pet.hunger = 8
+        pet.feed()
+        expect(pet.hunger).toEqual(0)
+      })
```

Comment:
This test isnt doing what it should
```
    it('it prevents hunger level from going below 0', () => {
        const pet = new Pet ('Bob');
        pet.hunger = 1;
        pet.feed();

        expect(pet.hunger).toEqual(0)
    })
```
If you update this test to this, it will fail due to an error in the logic being tested. Additionally we should add another test to make sure the hunger value is changed by the right amount when it isnt expected to go below 0:
```
    it("reduces the hunger value by the correct amount", () => {
        const pet = new Pet("Bob");
        pet.hunger = 8;
        pet.feed();

        expect(pet.hunger).toEqual(5);
    });
```